src/main/java/yeelp/distinctdamagedescriptions/integration/tic/tinkers/capability/TinkerDamageDistribution.java

```diff
@@ -55,7 +56,10 @@ public <T> T getCapability(Capability<T> capability, EnumFacing facing) {
 	@Override
 	public IDamageDistribution update(IProjectile owner) {
 		if(owner instanceof EntityProjectileBase) {
-			DDDAPI.accessor.getDamageDistribution(((EntityProjectileBase) owner).tinkerProjectile.getItemStack()).ifPresent((dist) -> this.setNewWeights(dist.getCategories().stream().collect(DDDBaseMap.typesToDDDBaseMap(() -> 0.0f, dist::getWeight))));
+			EntityProjectileBase epb = ((EntityProjectileBase) owner);
+			if(epb.tinkerProjectile != null) {
```

Comment:
This does solve the problem, which for starters, thanks! But, in what context is the tinker projectile null, and is that a context we need to care about?

I wouldn't want something along the lines of tinker arrows randomly inflicting just piercing damage because sometimes this is null.

Reply:
Its constructor has three versions, and only one of it initialize tinkerProjectile field. But I didn't figure out why this fix works after skipping the distribution applying

Reply:
Interesting. I think I'd like to know why this fix works before merging, just to prevent potential future problems

Reply:
So I debug this with Tinker and found this method was access at Entity's attachCapability event, which before tinkerProjectile initialized, so tinkerProjectile will always be null. The damage distribution still get applied though, can't figure out why